### ISSUE TITLE
log flood

### ISSUE THREAD

**Issue by xificurC** [juxt/pack.alpha#77]
When building with [slf4j-timbre](https://github.com/fzakaria/slf4j-timbre) on the classpath all logs get routed through [timbre](https://github.com/ptaoussanis/timbre), at debug level, which starts flooding me with debug messages like this one below. This might not be a pack issue, but if you have any idea how to tame this monster and e.g. set up info level debugging I'd be really happy.
```
20-03-17 12:58:06 * DEBUG [org.apache.http.wire:73] -  << "eKu:[0x18]vy[0xa0][0xd7][0xe1]4[0x92][0xd6]5z[0xe6]d[0xaa]dq[0xc6][0x83][0xbc]D'i[0xef][0x16][0xc]-[0xd7]=<[0xfa]tP[0xf0][0xa4][0x8e][0x17][0xb4][0x9c]6[0x85][0x9][0xc8]w[0x4][0xf8][0x83]qe[0xa6]#[0x8a][0x8b]|X[0xe9][0xb0]m[0x15][0xe4][0x6]}[0xbc][0xb9]ki[0x16][0x16]v[0xb]v[0xbd][0x1a][0x18][0x15][0x8c][0xdb][0xc2]v[0xe6][0x99];4[0xeb][0xad]VH [0x80]4[0xee]k[0x87][0xc]@h[0xd4][0xf7][0xf1][0xb0]0[0xef]T![0xb8][0xde]yS[0xeb][0xab]+[0xe6][0x9e][0xae][0xbd]q[0xf6]X[0x1f][0xbf][0xf5]C/[0x4][0xb2][0xe8][0xa5][0xc1]r[0xb9][0xf0]k[0xc6][0xf0]ni0[0xd2][0x97]P[0xf7][0xfc]7fM[0x2]T[0xb3],M|O[0x84]Zr[0x9e]ZNC[0xb6][0x1c][0xd9][0x9c][0xc6][0xed]|[0x87][0xa5]j[0xdc][0xff]V:[0x80][0xbe]v[0xe9][0xfc][0xc2][0xcc][0xf0][0xf8][0xfd][0x7][0xd0][0x1b]\[0xd1][0xf7][0xd7]$[0x1c][0xdb]\$[0xb]_[0xf4][0xeb]D[0xc4][0xb7]T[0xa4][0xfa]$[0xf0]/Y[0xd0]W[0xc0]r[0xb0][0xef][0x1d]7[0xc]=[0xbb]R[0xf3][0x1d]%[0xff]K[0xc3][\r][0xdf][0xe8][0xac][0xfc][0x85]iJ[0x1f]z[0xfa][0x1d][0xbb][0x84]/[0xca][0xb9]q`[0x15]n[0xe7]C9B[0x9d][0xc9][0xf9][0xd5]d[0x91][0x95]#[0xee][0x92]w[0xdd][0xe5][0x8f][0x8f][0xf3]P[0xe7][0x8a][0xb]0][0xf][0x9][0xca][0xe2][0xa1][0xd7]4d[0xd4][0xfc][0xae]s[0xd1]nr:n[0xb8]q[0xa2][0xcb]"[0xfd][0xdf]P[0x16][0xf8]O0[0xfb]OP[0x16][0xf8]3[0xcc]Z[0xbb]\|C[0xd9][0xe5][0x8f]([0xb]|[0x83]Y[0x99][0x93][0xff][0x17][0x82]([0x5][0x86]5[0x86]/[0x1][\r]p[0xa2][0x93][0x97][0xe5][0x1a][0xc4]bv[0x93][0xeb][0xe1][0x99][[0x2]E[0xdc][0x96],[0xc6].[0xe5][0x12][0x8e][0xe9][0x1d]K[0xdd][0xea]i[0xad][0xcc][0xaf])2.[0xc4][0x1a][0xb7]9[0x12][0xaf][0xb4]C=y[0x8b][0x9d]{[0x0][0x2][0xab][0xbb][0xae][0xcd] k[0xbf]{'[0xe]7[0x15][0xa2][0x9b]}|I#xO[0x95]'8[0xb0][0x5][0x1b][0x96][0xfe][0xfb][0xdd][0xe7][0xfb]x>e[0x98][0x1][0x97][0xdd][0xf5][0xa6]`O[0x8d]`U[0x5][0x6]0[0x95][0xcd][0x6]/[0xe1]V[0x83][0xcc][0xb]K[0xf9][0x1b][0x87]a[0x81][0x13]p!3@[0x11][0x98][0xc][0x1d][0xaf][0xce]o[0x88][0xe2][0xec][0x18][0x83]R{[0xc2]%[0xdd][0xf0][0xd6]l[0x91][0x84][0x88][0x9a][0xfa][0x8]my[0xe0][0xd5],[0xb5]"w[0xa2]V[7[0xd5];[0x9c][0xea]|[0xf7][0xbb][0xa9][0xda][0xaa][0xdc][0xe1][0xbb][0xb0][0xe1]R[0xde][0xf7][0xca][0x1d][0xbd]u[0x85][0x1a][0xee]Yl5[0xe3][0xcc]j[0x1a]2[0xae][0xae][0x1d][0xb3][0x13][0xc8][0x3][0xce]v{[0xdd]&[0xf3]u[0xa7][0xba]'[0xa6][0x14]5[0xa1][0xc6]S~Do[0x84]<[0x17][0xf3]8[0x9b][0x92]:[0xc4][0xc][0xc6][0xe2][0xa8]+[0x5][0x1f]t[0xac][0xa2]Q[0x9]kI[0xf7][0xe8][0xc9]1[0xc3][0x5]w[0xc0][0x0][0xa3][0x90]h[0xde][0xe5]3[0x82]B[0xdb][0xe9]b[0xbd][0xf7]X[0xfd][0x89][0x8d][0xca]\N[0xf4][0x9a][0xc0]Q[0xe3][0xdf][0x8c][0xc3]Od[0xa4]R`[0x8e]x[0xba][0xd5]s'yZ[0xdf]x(<![0xa4][0x2][0xe6][0x83][0xb0][0x98]v[0xca][0xa9][0xf1][0xf1][0xf0][0xd1] [0xa6]^oQ[0xec]rl&[0xc0][0xe0]V[0xbb][0xd9][0xf6][0xcc][0x1d] [\r].<[0xf4][0x9][0x18][0x12]m&?[0xf2][0xe4]i[0xfa][0x9c]g[0xe2]7[0xd9][0xc][0x0]3[0xf6][0x89]Tf[0x12][0xf7]0[0xce]c[0xc9])5[0x1c][0xf9][0xb5]U[0x1b]u[0xf1][0x19][0xbb]h[0x8a][[0xa5][0x16]K;[0xef]uz*[0x9c][0x11][0x86][0xd9][0xa3][0x92][0xbc][\r]M[0x88][0xfb]ak[0xc5][0xf3][0x2][0xdc][0x8d][0xb0][0xd8]|[0xdc][0x9][0xbc][0xbe][0xfb]k[0xbc] [0x84][0x1f]:[0xd4][0xd5]j[0xac][0xe7][0x9f][\n]"
```

**Comment by SevereOverfl0w** [comment]
This would be fixed by using :deps instead of :extra-deps.

**Comment by xificurC** [comment]
I'm not sure what you mean, I'm using :deps. This is my current deps.edn

```clojure
{:deps {org.clojure/clojure                     {:mvn/version "1.10.1"}

        ;; DBS
        org.mongodb/mongodb-driver-sync         {:mvn/version "3.10.1"}
        seancorfield/next.jdbc                  {:mvn/version "1.0.2"}
        com.microsoft.sqlserver/mssql-jdbc      {:mvn/version "7.2.2.jre11"}
        com.zaxxer/HikariCP                     {:mvn/version "3.3.1"}
        org.xerial/sqlite-jdbc                  {:mvn/version "3.28.0"}
        io.replikativ/datahike                  {:mvn/version "0.2.1"}

        ;; JODA - only because of groovy jobs..
        joda-time/joda-time                     {:mvn/version "2.10"}

        ;; UTILS
        com.rpl/specter                         {:mvn/version "1.1.2"}
        net.openhft/zero-allocation-hashing     {:mvn/version "0.10.1"}
        metosin/malli                           {:mvn/version "0.0.1-SNAPSHOT"}
        integrant                               {:mvn/version "0.7.0"}
        integrant/repl                          {:mvn/version "0.3.1"}
        primitive-math                          {:mvn/version "0.1.6"}
        io.micrometer/micrometer-core           {:mvn/version "1.3.2"}
        org.clojure/java.jmx                    {:mvn/version "1.0.0"}
        org.apache.commons/commons-lang3        {:mvn/version "3.9"}

        ;; CONVERTERS
        org.clojure/data.codec                  {:mvn/version "0.1.1"}
        metosin/jsonista                        {:mvn/version "0.2.5"}
        ;; cheshire                                {:mvn/version "5.8.1"}
        ;; org.clojure/data.xml                    {:mvn/version "0.2.0-alpha6"}
        ;; org.clojure/data.csv                    {:mvn/version "0.1.4"}

        ;; GROOVY
        org.codehaus.groovy/groovy              {:mvn/version "3.0.2"}

        ;; LOGGING
        com.taoensso/timbre                     {:mvn/version "4.10.0"}
        ;; fixes https://github.com/AvisoNovate/pretty/issues/53
        io.aviso/pretty                         {:mvn/version "0.1.37"}
        com.fzakaria/slf4j-timbre               {:mvn/version "0.3.14"}

        ;; REPL
        cider/cider-nrepl                       {:mvn/version "0.24.0"}
        }
 :jvm-opts ["-Djdk.attach.allowAttachSelf"
            "-XX:+UnlockDiagnosticVMOptions"
            "-XX:+DebugNonSafepoints"
            "-XX:-OmitStackTraceInFastThrow"]
 :aliases {:dev {:extra-deps {criterium {:mvn/version "0.4.5"}
                              com.clojure-goes-fast/clj-java-decompiler {:mvn/version "0.3.0-SNAPSHOT"}
                              com.clojure-goes-fast/clj-async-profiler {:mvn/version "0.4.0"}
                              expectations/clojure-test {:mvn/version "1.1.1"}
                              org.clojure/test.check {:mvn/version "0.10.0-alpha4"}}
                 :extra-paths ["classes" "test"]}
           :pack {:extra-deps {pack/pack.alpha {:git/url "https://github.com/juxt/pack.alpha.git"
                                                :sha "c70740ffc10805f34836da2160fa1899601fac02"}}
                  :main-opts ["-m"]}}}
```

**Comment by SevereOverfl0w** [comment]
For pack, you are using :extra-deps. Instead you can use :deps.

**Comment by xificurC** [comment]
Ah, I see, didn't think of that since the auto-installer picked this setup. Will try once I'm on my notebook again, thanks. Is there any way to just "run some code" before a pack command fires off?

**Comment by SevereOverfl0w** [comment]
There isn't. For now I would recommend running your own program first, or calling the main with arguments.

One day I'd like an API to solve that use case.

**Comment by viesti** [comment]
On the API thoughts, just ran into [badigeon](https://github.com/EwenG/badigeon), which seems to be a build _library_, so it has an API. I've been liking that pack.alpha kind of suites on being a set of cli commands to run separately, or at least, that fact made it fast to hack Jib support for it :)

Haven't really had any deep thoughts, but just throwing the idea into the air: could badigoen be the library that pack would call into, even? I was thinking if badigeon would have jib support also (maybe contribute it there).

Anyhows, really misusing this thread for API design, should talk about it elsewhere, maybe :)

**Comment by SevereOverfl0w** [comment]
Fixed by 0eff96cd1eb7c5810d4f8fdd8d7236b2df23505b
